### PR TITLE
feat: move data fetching to Algolia

### PR DESCRIPTION
**Pull request by dzbo**
### Description

- this is feature branch containing all PR's for the switch. We want to make 1 release with whole rework.

### CODE REVIEW

File: utils/fetchLsp8Metadata.ts [hunk 1]
```diff
@@ -188,5 +181,13 @@ const getLsp8Metadata = async (
     })
   })
 
-  return validateLsp4MetaData(lsp8Metadata)
+  console.debug('LSP8 metadata', {
```
Comment (doubleppereira) [inline utils/fetchLsp8Metadata.ts:184]: Remove this?

Comment (dzbo) [reply utils/fetchLsp8Metadata.ts:184]: I would like to keep debug logs at least for some time as it help while investigating issues.

**Review by doubleppereira — APPROVED** [review]
Did a quick run thought and LGTM but maybe someone with more context would be better to review this

**Review by richtera — APPROVED** [review]
LGTM. I think caching will need to be adjusted a little more. We should investigate into that. For example a tokenId should be cached, but a balance needs to be pulled more often.

**Pull request merged**